### PR TITLE
Adding layers docs

### PR DESCRIPTION
**Pull request by daniel-statsig**

### CODE REVIEW

File: docs/layers/introduction.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: Introduction
+sidebar_label: Introduction
+slug: /layers
+---
+
+# What are Layers?
+Layers are a way of performing experiments with out requring continued updates to your codebase. They are a 'write once' solution to experimentation. Layers work with Experiments to run tests on product features, and allow shared parameters between Experiments.
```
Comment (alex-statsig) [inline docs/layers/introduction.md:8]: spelling nit: "without requiring"

Comment (alex-statsig) [reply docs/layers/introduction.md:8]: also maybe "write code once"

File: docs/layers/introduction.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## When to use Layers
+Similar to our Experiments product, Layers are ideal for running A/B or A/B/n experiments. There differentiator is that you can avoid rewritting code between experiment runs.
```
Comment (alex-statsig) [inline docs/layers/introduction.md:11]: spelling nit: Their, and rewriting

File: docs/layers/introduction.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+title: Introduction
+sidebar_label: Introduction
+slug: /layers
+---
+
+# What are Layers?
```
Comment (tore-statsig) [inline docs/layers/introduction.md:7]: Layers already existed as a way to isolate experiments from eachother.  So we definitely need to call that out as well, as that functionality remains (and is another major reason to use them)

File: docs/layers/introduction.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Layers are a way of performing experiments with out requring continued updates to your codebase. They are a 'write once' solution to experimentation. Layers work with Experiments to run tests on product features, and allow shared parameters between Experiments.
+
+## When to use Layers
+Similar to our Experiments product, Layers are ideal for running A/B or A/B/n experiments. There differentiator is that you can avoid rewritting code between experiment runs.
+
+
+## Layers vs Experiments
```
Comment (alex-statsig) [inline docs/layers/introduction.md:14]: Maybe "Layers vs Standalone Experiments" or something? Just want to make it clear that you still use experiments. Maybe it's fine though

File: docs/layers/introduction.md (hunk 2 above)
Comment (tore-statsig) [inline docs/layers/introduction.md:11]: I would not introduce layers as similar to the experiments product.  Its a concept that isolates experiments, but you still run a/b/n tests with the Experiments product

File: docs/layers/introduction.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+# What are Layers?
+Layers are a way of performing experiments with out requring continued updates to your codebase. They are a 'write once' solution to experimentation. Layers work with Experiments to run tests on product features, and allow shared parameters between Experiments.
+
+## When to use Layers
+Similar to our Experiments product, Layers are ideal for running A/B or A/B/n experiments. There differentiator is that you can avoid rewritting code between experiment runs.
+
+
+## Layers vs Experiments 
+Why would you want to use Layers? Let's say you have an experiment that changes the color of the checkout button in your purchase flow. This code might look something like this:
```
Comment (tore-statsig) [inline docs/layers/introduction.md:15]: This isnt even a question.  You need an experiment either way, a layer does not run a test without an experiment

File: docs/layers/js-tutorial.mdx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+slug: /layers/js-tutorial
+---
+
+# Using a Layer in Javascript
+
+This example shows how to setup a Layer with multiple parameters and then run multiple experiments with different combinations of those parameters.
+
+## Create a Layer
+
+First, create a new layer by logging into the [Statsig Console](https://console.statsig.com), navigate to Experiments, select the Layers tab and then select Create New.
```
Comment (tore-statsig) [inline docs/layers/js-tutorial.mdx:13]: layer is sometimes capitalized, sometimes not throughout this.  I think we decided not to capitalize these things (Anu can confirm)

File: docs/layers/js-tutorial.mdx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+In the dialog that appears, you can enter the Name, Type and Default Value for your parameter.
+
+![Screen Shot 2022-03-21 at 10 13 51 AM](https://user-images.githubusercontent.com/95646168/159327612-b0f09c4c-9d47-48d0-972b-e1d27bc3d3f2.png)
+
+
+## Setup Experiments
+
+We can now reference our Layer during Experiment creation, allowing us to pull in these parameters and share them across Experiments.
+
+When creating and Experiment, expand the Advanced options, check Add Layer, and select your Layer from the layer dropdown.
```
Comment (alex-statsig) [inline docs/layers/js-tutorial.mdx:41]: spelling nit: "When creating an Experiment"

Comment (alex-statsig) [reply docs/layers/js-tutorial.mdx:41]: Also I added a new flow for this; on the layer's page, you can click "Create new Experiment in Layer". That avoids them having to go into the Advanced section and find their layer.

Fine with leaving docs like this though, just an FYI

File: docs/layers/js-tutorial.mdx [hunk 8]
```diff
@@ -0,0 +1,71 @@
+---
+title: Tutorial
+sidebar_label: Example in Javascript
+slug: /layers/js-tutorial
+---
+
+# Using a Layer in Javascript
+
+This example shows how to setup a Layer with multiple parameters and then run multiple experiments with different combinations of those parameters.
```
Comment (tore-statsig) [inline docs/layers/js-tutorial.mdx:9]: Id add a bit more motivation here - real world examples help a lot.

File: docs/layers/js-tutorial.mdx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+## Create a Layer
```
Comment (tore-statsig) [inline docs/layers/js-tutorial.mdx:11]: This whole guidance seems really good, but all belongs under a "Setup" category, likely

File: docs/layers/js-tutorial.mdx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+On the Layer page, select Add Parameter. 
+
+![Screen Shot 2022-03-21 at 10 13 30 AM](https://user-images.githubusercontent.com/95646168/159327580-837ab0ef-5449-4cb7-b09b-41f3bfa93e63.png)
+
+
+In the dialog that appears, you can enter the Name, Type and Default Value for your parameter.
+
+![Screen Shot 2022-03-21 at 10 13 51 AM](https://user-images.githubusercontent.com/95646168/159327612-b0f09c4c-9d47-48d0-972b-e1d27bc3d3f2.png)
+
```
Comment (alex-statsig) [inline docs/layers/js-tutorial.mdx:35]: Maybe mention "You can also add more layer parameters later from the Experiment page as you come up with new ones"

File: docs/layers/js-tutorial.mdx [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+On the Experiment page, add your Layer parameters to the Experiment groups by selecting Add Parameter.
+
+![Screen Shot 2022-03-21 at 10 23 31 AM](https://user-images.githubusercontent.com/95646168/159329356-97787500-ac38-4a66-ad88-e5980ed2d992.png)
+
+In the dialog that appears, select your existing Layer parameter from the dropdown menu.
+
+![Screen Shot 2022-03-21 at 10 20 10 AM](https://user-images.githubusercontent.com/95646168/159328907-aa825add-e328-4407-b6d9-f2c269abfc2d.png)
+
+Now, when this Experiment is running, Users that are put into these groups will get the associated values.
+
```
Comment (alex-statsig) [inline docs/layers/js-tutorial.mdx:55]: Maybe worth mentioning "For any parameter not added to your experiment, the Layer defaults will be used"

**Review by alex-statsig — APPROVED** [review]
lgtm overall, couple of spelling nits and thoughts

File: docs/layers/introduction.md (hunk 5 above)
Comment (alex-statsig) [reply docs/layers/introduction.md:15]: I think it is fair to ask "Why would you want to *add layers* to your experiments?" or "run your experiments in layers" or something, but agreed we shouldn't make it sound like it replaces Experiments

**Pull request merged**